src/main/resources/avro/common.avdl

```diff
@@ -18,33 +337,116 @@ enum Strand {
+member.
 */
-// TODO: Add support here for universally unique base IDs.
 record Position {
   /**
-  The name of the reference (or, more technically, the scaffold) in whatever
-  reference set is being used. Does not generally include a "chr" prefix, so for
-  example "X" would be used for the X chromosome.
+  The name of the reference sequence in whatever reference set is being used.
+  Does not generally include a "chr" prefix, so for example "X" would be used
+  for the X chromosome.
+
+  If `sequenceId` is null, this must not be null.
+  */
+  union { null, string } referenceName = null;
+
+  /**
+  The ID of the sequence on which the `Position` is located. This may be a
+  `Reference` sequence, or a novel piece of sequence associated with a
+  `VariantSet`.
+
+  If `referenceName` is null, this must not be null.
+
+  If the server supports the "graph" mode, this must not be null.
   */
-  string referenceName;
+  union { null, string } sequenceId = null;
 
   /**
-  The 0-based offset from the start of the forward strand for that reference.
-  Genomic positions are non-negative integers less than reference length.
+  The 0-based offset from the start of the forward strand for that sequence.
+  Genomic positions are non-negative integers less than sequence length.
   */
   long position;
 
   /**
-  Strand the position is associated with.
+  Strand the position is associated with. `POS_STRAND` represents the forward
+  strand, or equivalently the left side of a base, and `NEG_STRAND` represents
+  the reverse strand, or equivalently the right side of a base.
   */
   Strand strand;
 }
 
 /**
+A `Segment` is a range on a sequence, possibly including the joins at the
+sequence's ends. It does not include base data. (The bases for a sequence are
+available through the `getSequenceBases()` API call.)
+
+In the sequence "GTGG", the segment starting at index 1 on the forward strand
+with length 2 is the "TG" on the forward strand. The length-2 segment starting
+at index 1 on the reverse strand is "AC", corresponding to the first two base
+pairs of the sequence, or the last two bases of the reverse complement.
+*/
+record Segment {
+  /**
+  The sequence ID and start index of this `Segment`. This base is always
+  included in the segment, regardless of orientation.
+  */
+  Position start;
+
+  /**
+  The length of this `Segment`'s sequence. If `start` is on the forward strand,
+  the `Segment` contains the range [`start.position`, `start.position` +
+  `length`). If `start` is on the reverse strand, the `Segment` contains the
+  range (`start.position` - `length`, `start.position`]. This is equivalent to
+  starting from the side indicated by `start`, and traversing through that base
+  out to the specified length.
+  */
+  long length;
+
+  /**
+  Start and end `Position`s where this `Segment` attaches to other sequences.
+  Note that the segmentId for start and end might not be the same. The
```

Comment:
"segmentId" doesn't appear to be defined anywhere, is it out-of-date terminology?